src/core_modules/capture-core/components/TopBarActions/TopBarActions.container.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import i18n from '@dhis2/d2-i18n';
+import { ActionButtons } from './TopBarActions.component';
+import { openNewRegistrationPageFromScopeSelector, openSearchPageFromScopeSelector } from './TopBarActions.actions';
+import {
+    resetAllCategoryOptionsFromScopeSelector,
+    resetProgramIdBatchAction,
+} from '../ScopeSelector/ScopeSelector.actions';
+import { resetProgramIdBase } from '../ScopeSelector/QuickSelector/actions/QuickSelector.actions';
+import { useReset, useSetOrgUnitId } from '../ScopeSelector/hooks';
+import { ConfirmDialog } from '../Dialogs/ConfirmDialog.component';
+import type { Props } from './TopBarActions.types';
+
+const defaultDialogProps = {
+    header: i18n.t('Unsaved changes'),
+    text: i18n.t('Leaving this page will discard the changes you made to this event.'),
+    confirmText: i18n.t('Yes, discard'),
+    cancelText: i18n.t('No, stay here'),
+};
+
+export const TopBarActions = ({
+    selectedProgramId,
+    selectedOrgUnitId,
+    isUserInteractionInProgress = false,
+    customActionsOnProgramIdReset = [],
+}: Props) => {
+    const [openStartAgainWarning, setOpenStartAgainWarning] = useState(false);
+    const dispatch = useDispatch();
+    const { reset } = useReset();
+    const { setOrgUnitId } = useSetOrgUnitId();
+    const dontShowWarning = () => !isUserInteractionInProgress;
```

Comment:
hey @simonadomnisoru, why do you invoke a function here? 

nit: I see you only refactor file without changing it, but will we have plan to clean up this later? I think a good pattern for the naming here would be `shouldNotShowWarning`? wdyt?

Reply:
Hey, 
I decided to use the `isUserInteractionInProgress` prop directly without storing it in any new variable. Let me know wdyt. 
Thanks!